### DISCUSSION TITLE
supabase js client storage info method 400 Bad Request

### DISCUSSION THREAD

**Discussion by chmoder** [supabase/supabase#30856] [Questions] [answered]
```TypeScript
const { data, error } = await supabase.storage
    .from('bucketId')
    .info(filePath);
```
generates the request:
GET `https://<project_id>.supabase.co/storage/v1/object/info/<bucket_id>/<filepath>`

This works locally, but after deploying it always returns **400 Bad Request**.  My CI/CD pipeline uses migrations so I think the policies are the same and have verified the SELECT policy is the same.

Any advice on how to use this method would be helpful, I am just trying to get the `contentType` of the file.

**Comment by GaryAustin1** [comment]
There is a bug it seems.   change the .from to `public/bucketId` if that is a public bucket...
https://github.com/supabase/storage-js/issues/209#issuecomment-2476633673

**Reply by chmoder** [reply]
I don't think my bucket is public. Is there another workaround?

**Reply by GaryAustin1** [reply]
I just searched the repository for `info`.  I don't think it is actually even documented yet though for the js client.
You may need to generate an issue in storage or storage-js if you don't see anything thing else there.

**Reply by GaryAustin1** [reply]
Well... I played with it some and this works for a non-public bucket....
`authenticated/bucketId`

**Reply by chmoder** [reply]
Thanks for all the effort.  I've been trying other things and now I think it has something to do with RLS policies and the way that `info` is querying for the `contentType` (etc).

```SQL
insert into storage.buckets
  (id, name)
values
  ('bucket_id', 'bucket_id');


create policy "storage-bucket_id-user-select-policy"
on storage.objects
for select
to authenticated
using (
  bucket_id = 'bucket_id' and
  (storage.foldername(name))[1] = (select auth.uid()::text)
);

create policy "storage-bucket_id-user-insert-policy"
on storage.objects
for insert
to authenticated
with check (
  bucket_id = 'bucket_id' and
  (storage.foldername(name))[1] = (select auth.uid()::text)
);

create policy "storage-bucket_id-user-update-policy"
on storage.objects
for update
to authenticated
using ( (select auth.uid()::text) = owner_id )
with check (
  bucket_id = 'bucket_id' and
  (storage.foldername(name))[1] = (select auth.uid()::text)
);

create policy "storage-bucket_id-user-delete-policy"
on storage.objects
for delete
to authenticated
using (
  bucket_id = 'bucket_id' and
  (storage.foldername(name))[1] = (select auth.uid()::text)
);
```

**Reply by GaryAustin1** [reply, marked as answer]
I would change your policy on select to public and bucket_id = 'x' and see if it works as a test.
If you don't meet select RLS it won't work.

I have actually run both the public bucket one and the private bucket one with select RLS for both and it worked for me on a Hosted Supabase instance with the latest supabase-js code.

**Reply by chmoder** [reply]
I deleted the policy and created a new with with public and it worked.  Then I deleted it again and reverted back to `authenticated`.   That worked as well.  So just to be safe I deleted all the related policies and recreated them and all is well.

I hope this isn't the case, but it seems like I may have done some out of band schema changes and the migrations didn't reflect the actual state in supabase.

**Reply by GaryAustin1** [reply]
Just to clarify you still have to "patch" the info call to get it to work even with the RLS fix?

**Reply by chmoder** [reply]
Good question.  It works with and without "authenticated" prefixing the bucket id.  Thank you @GaryAustin1!

**Reply by GaryAustin1** [reply]
That is odd as it won't work for me without authenticated.
Are you using next.js by any chance?  If so caching it does might be messing with your testing of changes serverside...

**Reply by GaryAustin1** [reply]
I guess caching would not make sense because when you change the path the url would change anyway.

**Reply by chmoder** [reply]
I am using `@supabase/supabase-js@2.46.1` in react.

**Reply by GaryAustin1** [reply]
I'm on 2.46.2.
I'll leave it to Supabase to sort as of now as you are running and I don't use it.
